# Cálculo - Tarefa 02

1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.  
2- Calcule essa área de forma analítica.

#### Soma de Riemann para usar de referência<br>
∫ 
1
2
​
 x 
3
 dx≈∑ 
i=1
n
​
 (x 
i
​
 ) 
3
 Δx

#### Será usado n = 4 para essa aproximação.

In [27]:
n = 50
a = 1
b = 2

#### Formula das larguras
Δx= (b-a) / n

#### Sendo assim fica:


In [28]:
delta_x = (b-a) / n
print(f'Δx = {delta_x}')

Δx = 0.02


#### Formula para os subintervalos, onde i é o numero de subintervalos
xi = a + i * Δx

In [29]:
xi = {}
i = 1

while i <= n:
    xi[f'x{i}'] = a + i * (delta_x)
    i = i + 1
print('Subintervalos de 1 a N, sendo denotado por i:')
xi


Subintervalos de 1 a N, sendo denotado por i:


{'x1': 1.02,
 'x2': 1.04,
 'x3': 1.06,
 'x4': 1.08,
 'x5': 1.1,
 'x6': 1.12,
 'x7': 1.1400000000000001,
 'x8': 1.16,
 'x9': 1.18,
 'x10': 1.2,
 'x11': 1.22,
 'x12': 1.24,
 'x13': 1.26,
 'x14': 1.28,
 'x15': 1.3,
 'x16': 1.32,
 'x17': 1.34,
 'x18': 1.3599999999999999,
 'x19': 1.38,
 'x20': 1.4,
 'x21': 1.42,
 'x22': 1.44,
 'x23': 1.46,
 'x24': 1.48,
 'x25': 1.5,
 'x26': 1.52,
 'x27': 1.54,
 'x28': 1.56,
 'x29': 1.58,
 'x30': 1.6,
 'x31': 1.62,
 'x32': 1.6400000000000001,
 'x33': 1.6600000000000001,
 'x34': 1.6800000000000002,
 'x35': 1.7000000000000002,
 'x36': 1.72,
 'x37': 1.74,
 'x38': 1.76,
 'x39': 1.78,
 'x40': 1.8,
 'x41': 1.82,
 'x42': 1.8399999999999999,
 'x43': 1.8599999999999999,
 'x44': 1.88,
 'x45': 1.9,
 'x46': 1.92,
 'x47': 1.94,
 'x48': 1.96,
 'x49': 1.98,
 'x50': 2.0}

#### Soma de Riemann para exemplo, usando N = 4
≈(1.25) 
3
 ×0.25+(1.5) 
3
 ×0.25+(1.75) 
3
 ×0.25+(2) 
3
 ×0.25

≈((x1 + delta_x) ** 3) + ((x2 + delta_x) ** 3) ... N, onde N está em 50 retângulos até alcançar o valor de B usando o delta X

In [30]:
i = 1
riemann = 0
while i <= n:
    riemann = riemann + (xi[f'x{i}'] ** 3) * delta_x
    i+=1
    
riemann

3.8203

### 1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.  

##### Ao usar 50 subconjuntos (50 retângulos) a integral tem o valor de 3.8203

### 2- Calcule essa área de forma analítica.

In [ ]:
Calcule essa área de forma analítica

In [45]:
a = 1
b = 2
exp = 3

'''A constante 4 no denominador vem da regra de potência para integrais. Quando integra x**3, 
adiciona 1 ao expoente e divide pelo novo expoente. Neste caso, ao integrar x**3, é adicionado 1 a 3, 
obtendo 4, e divide pelo novo expoente 4.'''
integral_a1_b2_x3 = (((b**(exp+1))) / (exp+1)) - (((a**(exp+1))) / (exp+1))
integral_a1_b2_x3


3.75

Portanto, a integral definida acima com os limites a e b e sua função, representa a área sob a curva de x³ no intervalo de 1 a 2, sendo 3.75.